mlc_llm/core.py

```diff
@@ -359,16 +359,16 @@ def mod_transform_before_build(
         mod
     )  # pylint: disable=not-callable
 
-    if "num_attention_heads" in config and "hidden_size" in config:
+    if config.num_attention_heads and config.hidden_size:
         max_seq_len = None
         if args.max_seq_len > 0:
             max_seq_len = args.max_seq_len
-        elif "max_sequence_length" in config:
-            max_seq_len = config["max_sequence_length"]
+        elif config.max_sequence_length:
```

Comment:
> I think #826 fixes the issue. Could you try drop all changes here and rebase on to the latest main?

It turns out that #826 is not a complete fix as found by @CharlieFRuan. This line need to updated with `hasattr` as well.

Reply:
Sorry I did not see this PR and made a separate one: https://github.com/mlc-ai/mlc-llm/pull/829.

I'll close that one as @sudeepag's should fix the issue.